src/TheThingsNetwork.cpp

```diff
@@ -219,14 +219,40 @@ bool TheThingsNetwork::personalize(const byte devAddr[4], const byte nwkSKey[16]
   return true;
 }
 
-bool TheThingsNetwork::join(const byte appEui[8], const byte appKey[16], bool restart, long int nbr_delay, int max_attempts) {
+bool TheThingsNetwork::join(long int nbr_delay, int max_attempts) {
   int nbr_attempts = 0;
 
   while (nbr_attempts++ != max_attempts) {
     delay(nbr_delay);
-    if (restart) {
-      reset();
+    String devEui = readValue(F("sys get hweui"));
+    String str = "";
+    str.concat(F("mac set deveui "));
+    str.concat(devEui);
+    sendCommand(str);
+    if (!sendCommand(F("mac join otaa"))) {
+      debugPrintLn(F("Send join command failed"));
+      continue;
+    }
+    String response = readLine(10000);
+    if (response != F("accepted")) {
+      debugPrint(F("Join not accepted: "));
+      debugPrintLn(response);
+      continue;
     }
+    debugPrint(F("Join accepted. Status: "));
+    debugPrintLn(readValue(F("mac get status")));
+    return true;
+  }
+  return false;
+}
+
+
+bool TheThingsNetwork::join(const byte appEui[8], const byte appKey[16], long int nbr_delay, int max_attempts) {
+  int nbr_attempts = 0;
+
+  while (nbr_attempts++ != max_attempts) {
+    delay(nbr_delay);
+    reset();
```

Comment:
If I call reset() only once, join() fails all the time for me on the second or third loop. It says there are no free channels available... Is that normal ?
```
Sending: mac set appeui with 8 bytes
Sending: mac set deveui 0004A30B001B4C9F
Sending: mac set appkey with 16 bytes
Sending: mac join otaa
Response is not OK: no_free_ch
Send join command failed
```